module/observations/helper.js

```diff
@@ -1772,21 +1773,41 @@ module.exports = class ObservationsHelper {
                 
                 let observationData = await this.observationDocuments({
                     _id : observationId,
-                },["_id","solutionId"]);
+                },["_id","solutionId","programId"]);
                 
                 let solutionData;
+                let programUsers;
+                let requestForPIIConsent;
                 if(observationData[0]){
 
                     solutionData = await solutionHelper.solutionDocuments({
                         "_id" : observationData[0].solutionId
 
                         },[
                             "allowMultipleAssessemts",
-                            "license"
+                            "license",
+                            "rootOrganisations"
                     ]);
+
+                    const query = { 
```

Comment:
@VISHNUDAS-tunerlabs can we have one function in programUsersHelper that take 2 arguements of userId and programId and returns requestForPIIConsent, and one more function which takes programId and userId return if user has joined program or not, then use those 2 functions everywhere instead of duplicating the same code everywhere in the service.

Reply:
@aks30 we have requestForPIIConsent programs not in programUsers. and to check program joined or not we can use programUsersDocuments function. we generalised the programUsersDocuments. if we introduce new function it will be doing the same.